external/fv3fit/tests/reservoir/test_reservoir_computing_model.py

```diff
@@ -54,19 +59,26 @@ def test_dump_load_preserves_matrices(tmpdir):
         input_coupling_sparsity=0,
     )
     reservoir = Reservoir(hyperparameters, input_size=input_size)
-    readout = generic_readout(
-        coefficients=np.random.rand(input_size, state_size),
+    readout = ReservoirComputingReadout(
+        coefficients=sparse.coo_matrix(np.random.rand(input_size, state_size)),
         intercepts=np.random.rand(input_size),
     )
-    predictor = ReservoirComputingModel(reservoir=reservoir, readout=readout,)
+    predictor = ReservoirComputingModel(
+        reservoir=reservoir, readout=readout, square_half_hidden_state=False,
+    )
     output_path = f"{str(tmpdir)}/predictor"
     predictor.dump(output_path)
 
     loaded_predictor = ReservoirComputingModel.load(output_path)
-    assert _sparse_allclose(loaded_predictor.reservoir.W_in, predictor.reservoir.W_in)
-    assert _sparse_allclose(loaded_predictor.reservoir.W_res, predictor.reservoir.W_res)
+    assert _sparse_allclose(
+        loaded_predictor.reservoir.W_in.todense(), predictor.reservoir.W_in.todense()
+    )
+    assert _sparse_allclose(
+        loaded_predictor.reservoir.W_res.todense(), predictor.reservoir.W_res.todense()
+    )
```

Comment:
Is there a need for `_sparse_allclose` if converting the inputs to dense matrices ahead of time?  Or I guess alternatively, what prompted the need to convert the inputs to dense matrices?

Reply:
Good catch, I removed the todense() conversion since directly comparing the sparse matrices in the objects is safest.